### PR TITLE
Docs: Cluster Allocation Filtering

### PR DESCRIPTION
**Pull request by joshuar**
Put more emphasis on the fact that multiple values can be specified
and move examples after explanation of settings.

Suggestion for #16997

### CODE REVIEW

File: docs/reference/modules/cluster/allocation_filtering.asciidoc [hunk 1]
```diff
@@ -27,35 +52,8 @@ NOTE: Shards will only be relocated if it is possible to do so without
 breaking another routing constraint, such as never allocating a primary and
 replica shard to the same node.
 
-Cluster-wide shard allocation filtering works in the same way as index-level
-shard allocation filtering (see <<index-modules-allocation>> for details).
```
Comment (dakrone) [inline docs/reference/modules/cluster/allocation_filtering.asciidoc]: Why remove the link to index-level allocation filtering? It was a useful link

Comment (joshuar) [inline docs/reference/modules/cluster/allocation_filtering.asciidoc]: That link should now be in the first paragraph (I don't think it shows in the diff) so still exists on the page.  I was just removing what now became duplicate text through this rearranging.

Comment (dakrone) [inline docs/reference/modules/cluster/allocation_filtering.asciidoc]: Gotcha, I see it now

**Comment by dakrone** [comment]
LGTM

**Comment by clintongormley** [comment]
LGTM

**Comment by dakrone** [comment]
@joshuar do you want to merge this?

**Comment by dakrone** [comment]
@joshuar ping again about merging this (it should be rebased probably beforehand)

**Comment by joshuar** [comment]
@dakrone yep, I'll rebase and update.

**Comment by joshuar** [comment]
Superceded by #20958.

**Pull request closed without merging**